File: java/SwimmingCompetition/SwimmingCompetitionServices/src/main/java/swimmingcompetition/services/ISwimmingCompetitionServices.java
package swimmingcompetition.services;


import swimmingcompetition.model.Challenge;
import swimmingcompetition.model.Organizer;
import swimmingcompetition.model.Participant;
import swimmingcompetition.model.Registration;
import swimmingcompetition.model.dto.ChallengeDTO;
import swimmingcompetition.model.dto.ParticipantDTO;

import java.util.List;

public interface ISwimmingCompetitionServices {
    void login(Organizer user, ISwimmingCompetitionObserver client) throws SwimmingCompetitionException;
    Organizer findAccount(String username, String password) throws SwimmingCompetitionException;
    ChallengeDTO[] findAllChallenges() throws SwimmingCompetitionException;
//    int getNumberOfParticipantsByChallenge(Challenge challenge) throws SwimmingCompetitionException;
    List<ParticipantDTO> findParticipantsByChallenge(Challenge challenge) throws SwimmingCompetitionException;
//    List<Challenge> findChallengesByParticipant(Participant participant) throws SwimmingCompetitionException;
    Participant addParticipant(String name, int age) throws SwimmingCompetitionException;
    void addParticipantToMoreChallenges(Participant participant, List<Challenge> challenges) throws SwimmingCompetitionException;
    void logout(Organizer user, ISwimmingCompetitionObserver client) throws SwimmingCompetitionException;
}

File: csharp/SwimmingCompetitionCS/services/ISwimmingCompetitionServices.cs
using System;
using System.Collections.Generic;
using model;

namespace services
{
    public interface ISwimmingCompetitionServices
    {
        void login(Organizer user, ISwimmingCompetitionObserver client);
        Organizer findAccount(String username, String password);
        ChallengeDTO[] findAllChallenges();
        // int getNumberOfParticipantsByChallenge(Challenge challenge);
        IList<ParticipantDTO> findParticipantsByChallenge(Challenge challenge);
        // IList<Challenge> findChallengesByParticipant(Participant participant);
        Participant addParticipant(String name, int age);
        void addParticipantToMoreChallenges(Participant participant, List<Challenge> challenges);
        void logout(Organizer user, ISwimmingCompetitionObserver client);
    }
}